File: Backend/deletecollections.py
from pymongo import MongoClient

# MongoDB connection
client = MongoClient("mongodb://localhost:27017/")  # Apne URI ke hisaab se update karein
db = client["electionAT"]

# Collections jo aapko **delete nahi** karni hai
collections_to_keep = ["AssemblyMaps", "Assemblypolygens", "Districtpolygens", "Parliamentpolygens", "users"]  # Yahan apni collections ka naam daalein jo bachaana hai

# Sabhi collections ka list
all_collections = db.list_collection_names()

# Delete karne waali collections
collections_to_delete = [col for col in all_collections if col not in collections_to_keep]

# Delete process
for col in collections_to_delete:
    db[col].drop()
    print(f"Deleted collection: {col}")

print("Finished deleting unwanted collections.")

File: Backend/delete_collections.js
const { MongoClient } = require('mongodb');

async function dropSelectedCollections() {
  const client = new MongoClient('mongodb://localhost:27017');

  try {
    await client.connect();
    const db = client.db('electionAT');

    // जिन collections को बचाना है
    const keepCollections = ['assemblies', 'divisions', 'districts', 'parliaments', 'users'];

    const collections = await db.listCollections().toArray();

    for (const coll of collections) {
      if (!keepCollections.includes(coll.name)) {
        try {
          await db.collection(coll.name).drop();
          console.log(`❌ Dropped collection: ${coll.name}`);
        } catch (err) {
          console.error(`Failed to drop collection ${coll.name}:`, err.message);
        }
      } else {
        console.log(`✅ Kept collection: ${coll.name}`);
      }
    }

    console.log('🎯 Finished dropping unwanted collections.');
  } finally {
    await client.close();
  }
}

dropSelectedCollections().catch(console.error);
